### PR TITLE
RFC: Successor for GraphQL Architect

### PR DESCRIPTION
**Pull request by tbwiss**
# Description
 
RFC for a successor of the [GraphQL Architect](https://grandstack.io/docs/graphql-architect-overview/#:~:text=GraphQL%20Architect%20is%20a%20graph,Neo4j%20Desktop%20Graph%20Apps%20Gallery).

### CODE REVIEW

File: docs/rfcs/rfc-009-graphql-ui-poc.md [hunk 1]
```diff
@@ -0,0 +1,103 @@
...
+    2. Show the generated schema if introspected
+3. Build Neo4j GraphQL schema
+4. Query Editor page
+    1. Write queries
+    2. Execute queries
+    3. Show results
+5. Logout - _Repeat_
+
+### Input
+
```
Comment (tbwiss) [inline docs/rfcs/rfc-009-graphql-ui-poc.md:36]: @tbwiss add a few sketches

File: docs/rfcs/rfc-009-graphql-ui-poc.md [hunk 2]
```diff
@@ -0,0 +1,111 @@
...
+Keybindings
+
+1. CTRL+ENTER - to run the query
+2. CTRL+SHIFT+P - to prettify the code
+3. _Nice to have_: Other common/basic ones
+
+
+### Sketch
+
+![GraphQL_UI_v4](https://user-images.githubusercontent.com/8817964/155305646-2adbf310-079e-4348-9536-93cfb4da5215.png)
```
Comment (tbwiss) [inline docs/rfcs/rfc-009-graphql-ui-poc.md:68]: @tbwiss change "introspect" to "generate typeDefs" or similar

**Comment by darrellwarde** [comment]
https://user-images.githubusercontent.com/476818/51567716-c00dfa00-1e4c-11e9-88f7-6d78b244d534.gif

Inspiration?

File: docs/rfcs/rfc-009-graphql-ui-poc.md [hunk 3]
```diff
@@ -0,0 +1,112 @@
+# GraphQL Graph App PoC
+
+## Problem
+
+The [GraphQL Architect](https://grandstack.io/docs/graphql-architect-overview/#:~:text=GraphQL%20Architect%20is%20a%20graph,Neo4j%20Desktop%20Graph%20Apps%20Gallery) for usage in the Neo4j Desktop is due to be replaced. As an initial step a proof of concept of a successor shall be established.
+
+## Proposed Solution
+
+_Name suggestions:_  
+Explorer
```
Comment (danstarns) [inline docs/rfcs/rfc-009-graphql-ui-poc.md:10]: I plus +1 'Explorer'

File: docs/rfcs/rfc-009-graphql-ui-poc.md [hunk 4]
```diff
@@ -0,0 +1,112 @@
...
+Keybindings
+
+1. CTRL+ENTER - to run the query
+2. CTRL+SHIFT+P - to prettify the code
+3. _Nice to have_: Other common/basic ones
+
+
+### Sketch
+
+![GraphQL_UI_v5](https://user-images.githubusercontent.com/8817964/155744938-2ec1b531-3b83-44e1-9fd0-371022150274.png)
```
Comment (danstarns) [inline docs/rfcs/rfc-009-graphql-ui-poc.md:69]: Nice!

**Review by danstarns — COMMENTED** [review]
How many packages do you see us adding to the mono repo?

File: docs/rfcs/rfc-009-graphql-ui-poc.md [hunk 5]
```diff
@@ -0,0 +1,112 @@
...
+Explorer  
+Architect  
+Editor  
+UI  
+Frame  
+(Already taken: Studio, Playground, GraphiQL)
+
+### User audience
+
+The `@neo4j/graphql` lib's user audience are full stack developers who want a GraphQL API quickly without worrying about the database. This UI can further include developers wanting to experiment in general as well as experiment with/edit the introspected schema.
```
Comment (angrykoala) [inline docs/rfcs/rfc-009-graphql-ui-poc.md:19]: Question: Could this tool also be beneficial for our own development of `@neo4j/graphql` ?

File: docs/rfcs/rfc-009-graphql-ui-poc.md [hunk 6]
```diff
@@ -0,0 +1,112 @@
...
+
+### Workflow (Graph App)
+
+1. Login page
+    1. Specify the database credentials (connectURL, username, password)
```
Comment (angrykoala) [inline docs/rfcs/rfc-009-graphql-ui-poc.md:24]: **Nice-to-have** (for future versions maybe): Store credentials in localStorage or similar?

File: docs/rfcs/rfc-009-graphql-ui-poc.md [hunk 7]
```diff
@@ -0,0 +1,112 @@
...
+5. _Nice to have_: [Explorer plugin](https://user-images.githubusercontent.com/476818/51567716-c00dfa00-1e4c-11e9-88f7-6d78b244d534.gif) (or similar) for/from GraphiQL. [Repo](https://github.com/OneGraph/graphiql-explorer)
+6. _Nice to have_: Store latest queries&params in window [localStorage](https://developer.mozilla.org/en-US/docs/Web/API/Window/localStorage)
+7. _Nice to have_: Docs (utilize from [GraphQL Playground](https://github.com/graphql/graphql-playground)?)
+8. _Nice to have_: Tabs for query input and parameters (similar to Apollo Studio)
+
+Keybindings
+
+1. CTRL+ENTER - to run the query
+2. CTRL+SHIFT+P - to prettify the code
+3. _Nice to have_: Other common/basic ones
```
Comment (angrykoala) [inline docs/rfcs/rfc-009-graphql-ui-poc.md:64]: Fancy!

File: docs/rfcs/rfc-009-graphql-ui-poc.md [hunk 8]
```diff
@@ -0,0 +1,112 @@
+# GraphQL Graph App PoC
+
+## Problem
+
+The [GraphQL Architect](https://grandstack.io/docs/graphql-architect-overview/#:~:text=GraphQL%20Architect%20is%20a%20graph,Neo4j%20Desktop%20Graph%20Apps%20Gallery) for usage in the Neo4j Desktop is due to be replaced. As an initial step a proof of concept of a successor shall be established.
+
```
Comment (angrykoala) [inline docs/rfcs/rfc-009-graphql-ui-poc.md:6]: It is not completely clear where this is going to be available. Will it be a local server, webpage, desktop app...?

Maybe it is worth defining here.

**Review by angrykoala — DISMISSED** [review]
Looking good

File: docs/rfcs/rfc-009-graphql-ui-poc.md (hunk 5 above)
Comment (tbwiss) [reply docs/rfcs/rfc-009-graphql-ui-poc.md:19]: Do you mean as an "alternative" (or replacement) to the `test-api.ts`? I had that in mind when writing the pitch but was hesitant to put it as a requirement as we were unsure about the exact user audience for the tool. We will discuss the user audience later today and I'll bring this up. Good thou that you mention it!

File: docs/rfcs/rfc-009-graphql-ui-poc.md (hunk 3 above)
Comment (darrellwarde) [reply docs/rfcs/rfc-009-graphql-ui-poc.md:11]: I also side with Explorer for the most part, apart from the fact that this application will also let you _edit_ the type definitions, which for me goes beyond "exploring"!

Comment (tbwiss) [reply docs/rfcs/rfc-009-graphql-ui-poc.md:11]: Yes I came to like the name as well yet sadly it's already taken (sorry that's one me!), https://www.apollographql.com/docs/studio/explorer/explorer/ 
We may have to go with a working name for this first iteration/PoC.

File: docs/rfcs/rfc-009-graphql-ui-poc.md (hunk 5 above)
Comment (tbwiss) [reply docs/rfcs/rfc-009-graphql-ui-poc.md:20]: We just had a chat with Helene and yes, we should definitely build a v1/PoC that is to our own benefit as well. It is likely that other dev's will like to use the app if we ourselves are (and want to) using it too.

**Pull request merged**